rom/dev/tests/test_image_validation.rs

```diff
@@ -0,0 +1,465 @@
+    let image = caliptra_builder::sign_image(&image_bundle).unwrap();
+    hw.upload_firmware(&image).unwrap();
+    let mut output = vec![];
+    let result = hw.copy_output_until_fatal_error(MANIFEST_MARKER_MISMATCH, &mut output);
+    assert!(result.is_err());
+}
+
+#[test]
+fn test_invalid_manifest_size() {
+    let (mut hw, mut image_bundle) = build_hw_model_and_image_bundle();
+    image_bundle.manifest.size = (core::mem::size_of::<ImageManifest>() - 1) as u32;
+    let image = caliptra_builder::sign_image(&image_bundle).unwrap();
+    hw.upload_firmware(&image).unwrap();
+    let mut output = vec![];
+    let result = hw.copy_output_until_fatal_error(MANIFEST_SIZE_MISMATCH, &mut output);
+    assert!(result.is_err());
+}
+
+#[test]
+fn test_preamble_zero_vendor_pubkey_digest() {
+    // [TODO]
+    // Need:
+    // 1. LifecycleState != Unprovisioned
+    // 2. Need ability to set the vendor_pub_key_hash fuse to 0.
+}
+
+#[test]
+fn test_preamble_owner_pubkey_digest_mismatch() {
+    // [TODO]
+    // Need ability to update the owner_pub_key_hash fuse.
+    let (mut hw, mut image_bundle) = build_hw_model_and_image_bundle();
+    image_bundle.manifest.preamble.owner_pub_keys.ecc_pub_key.x[0] = 0xBAADF00D;
+    let image = caliptra_builder::sign_image(&image_bundle).unwrap();
+    hw.upload_firmware(&image).unwrap();
+    let mut output = vec![];
+    let _result = hw.copy_output_until_fatal_error(OWNER_PUB_KEY_DIGEST_MISMATCH, &mut output);
+    // [TODO] assert!(result.is_err());
+}
+
+#[test]
+fn test_preamble_vendor_pubkey_revocation() {
+    // [TODO] Need ability to update the fuse_key_manifest_pk_hash_mask fuse.
+}
+
+#[test]
+fn test_header_verify_vendor_sig_zero_pubkey() {
+    let (mut hw, mut image_bundle) = build_hw_model_and_image_bundle();
+    let vendor_ecc_pub_key_idx = image_bundle.manifest.preamble.vendor_ecc_pub_key_idx as usize;
+    let mut output = vec![];
+
+    // Set ecc_pub_key.x to zero.
+    let ecc_pub_key_x_backup =
+        image_bundle.manifest.preamble.vendor_pub_keys.ecc_pub_keys[vendor_ecc_pub_key_idx].x;
+    image_bundle.manifest.preamble.vendor_pub_keys.ecc_pub_keys[vendor_ecc_pub_key_idx]
+        .x
+        .fill(0);
+    let image = caliptra_builder::sign_image(&image_bundle).unwrap();
+    hw.upload_firmware(&image).unwrap();
+
+    let result = hw.copy_output_until_fatal_error(VENDOR_PUB_KEY_DIGEST_INVALID_ARG, &mut output);
+    assert!(result.is_err());
+
+    // Set ecc_pub_key.y to zero.
+    image_bundle.manifest.preamble.vendor_pub_keys.ecc_pub_keys[vendor_ecc_pub_key_idx].x =
+        ecc_pub_key_x_backup;
+    image_bundle.manifest.preamble.vendor_pub_keys.ecc_pub_keys[vendor_ecc_pub_key_idx]
+        .y
+        .fill(0);
+    let image = caliptra_builder::sign_image(&image_bundle).unwrap();
+    hw.upload_firmware(&image).unwrap();
+    let result = hw.copy_output_until_fatal_error(VENDOR_PUB_KEY_DIGEST_INVALID_ARG, &mut output);
+    assert!(result.is_err());
```

Comment:
result.is_err() is also true when somebody calls ExitCtrl::exit(1). I think we need this assertion to be stricter.